hazelcast/src/main/java/com/hazelcast/internal/partition/impl/MigrationListenerAdapter.java

```diff
@@ -33,21 +35,21 @@
     }
 
     @Override
-    public void onEvent(MigrationEvent migrationEvent) {
+    public void onEvent(ReplicaMigrationEvent event) {
 
-        final MigrationStatus status = migrationEvent.getStatus();
-        switch (status) {
-            case STARTED:
-                migrationListener.migrationStarted(migrationEvent);
-                break;
-            case COMPLETED:
-                migrationListener.migrationCompleted(migrationEvent);
+        switch (event.getPartitionId()) {
```

Comment:
Looks like it should be `switch (event.getMigrationState())`

Reply:
Unfortunately this is a hacky way I used to fire two different events. I used negative partition-ids to indicate `MIGRATION_STARTED ` and `MIGRATION_FINISHED`.